packages/cli/src/lib/setup/setupInstall.ts

```diff
@@ -947,34 +988,30 @@ function Install(options) {
                 : new RegExp(`^system\\.adapter\\.${adapter}\\.\\d+$`);
 
         try {
-            const doc = await objects.getObjectView('system', 'instance', {
+            const doc = await this.objects.getObjectView('system', 'instance', {
                 startkey: `system.adapter.${adapter}${instance !== undefined ? `.${instance}` : ''}`,
                 endkey: `system.adapter.${adapter}${instance !== undefined ? `.${instance}` : ''}\u9999`
             });
 
             // add non-duplicates to the list (if instance not given -> only for this host)
             const newObjIDs = doc.rows
                 // only the ones with an ID that matches the pattern
-                .filter(row => row && row.value && row.value._id)
-                .filter(row => instanceRegex.test(row.value._id))
+                .filter(row => row?.value && row.value._id)
+                .filter(row => row?.value && instanceRegex.test(row.value._id))
                 // if instance given also delete from foreign host else only instance on this host
                 .filter(row => {
-                    if (
-                        instance !== undefined ||
-                        !row.value.common ||
-                        !row.value.common.host ||
-                        row.value.common.host === hostname
-                    ) {
+                    if (instance !== undefined || row?.value?.common.host === hostname) {
```

Comment:
This logic was not equivalent (ignored the case where some part of row.value.common.host) is undefined. I changed it below.

Reply:
Changed it at which line? Yes, it differs a bit as far as I see if `hostname` is `undefined` and `common.host` is undefined it will now execute, previously not. But hostname is always string, so should be fine. Or what do you mean?

Reply:
Found it and added it as proposed, however I still think it is redundant

Reply:
As I understood it:
a) If row.value.common.host (or any of the parent objects) are **not** set, take the object
b) If row.value.common.host equals the given hostname, take the object

The new logic would only take the object in case b). This might be okay or intended, but I only looked at this logic and that part was different. Maybe a more verbose and less-broken-english comment should explain the desired behavior then.